hamilton/graph_types.py

```diff
@@ -203,5 +203,6 @@ def version(self) -> str:
         Node hashes are in a sorted list, then concatenated as a string before hashing.
         To find differences between dataflows, you need to inspect the node level.
         """
-        sorted_node_versions = sorted([n.version for n in self.nodes])
+        sorted_nodes = sorted(self.nodes, key=lambda n: n.name)
+        sorted_node_versions = [n.version if n.version else "" for n in sorted_nodes]
```

Comment:
I wonder if we want to fall back on the node name? E.G. if adding a materializer will make it the same hash?

Reply:
We'll that's the question. IMO, `HamiltonGraph.version` should return a hash equivalent to the state of the graph, so different materializers, different graph hash.

The idea is that by sorting by name, your `None` value are in the same position before being converted to `""`. Since we're concatenating everything to a string after, it should be equivalent to sorting node hashes and `""`.

Reply:
Yep, agreed. If you use a "", however, there are edge-cases in which it could be the same if it changes in some way? E.G. add a third matierializer that goes right after the second?

Also, the position of `""` doesn't matter, it won't effect it at all, right? E.G. no matter where the materializer are, an empty string has no impact. So won't this be the same regardless of where they end up?